### PR TITLE
Product Backlog Item 232670: SC/VLE, VR Java: Improve usability by ad…

### PR DESCRIPTION
**Pull request by dj4452**
…ding pause functionality.: #232670

### CODE REVIEW

**Comment by dj4452** [comment]
Added Review comment

File: src/uk/ac/open/tabapplet/TabApplet.java [hunk 1]
```diff
@@ -280,6 +282,23 @@ private void ditchFocus(boolean forward)
...
+	 * @param ispause boolean 
+	 **/
+	public void isPause(boolean isPause)
+	{
+		this.pauseFlag = isPause;
+		// Tell JavaScript to ditch focus for this applet id
+		String js = "isPause("+this.pauseFlag+");";
+		if(VERBOSE)
+		{
+			System.err.println("[uk.ac.open.tabapplet.TabApplet] " + focusHackId +
```
Comment (sammarshallou) [inline src/uk/ac/open/tabapplet/TabApplet.java]: This doesn't look like correct debugging text (copy/paste).

File: src/uk/ac/open/tabapplet/TabApplet.java [hunk 2]
```diff
@@ -30,6 +30,8 @@
 
 	private boolean internalIgnoreFocusChange;
 
+	private boolean pauseFlag = false;
```
Comment (sammarshallou) [inline src/uk/ac/open/tabapplet/TabApplet.java]: I don't think this private variable is really used any more - could it be removed?

File: src/uk/ac/open/tabapplet/TabApplet.java [hunk 3]
```diff
@@ -280,6 +282,23 @@ private void ditchFocus(boolean forward)
 		evalJS(js);
 	}
 
+	/**
+	 * To check if VR is paused
+	 * @param ispause boolean 
+	 **/
+	public void isPause(boolean isPause)
```
Comment (sammarshallou) [inline src/uk/ac/open/tabapplet/TabApplet.java]: The TabApplet class is supposed to be a generic one to deal with all applets you can tab around... Could this code be moved into the more specific applet class just for the voice recorder, I guess it is VoiceRecorderApplet or something?

File: build.xml [hunk 4]
```diff
@@ -197,13 +192,15 @@ http://forum.java.sun.com/thread.jspa?threadID=174214
 			<include name="uk/ac/open/embeddedrecorder/**"/>
     	<exclude name="uk/ac/open/embeddedrecorder/EmbeddedRecorderApp.java"/>
 		<manifest>
-			<attribute name="Permissions" value="{jarpermissions}"/>
-			<attribute name="Caller-Allowable-Codebase" value="{allowablecode}"/>
+			<attribute name="Permissions" value="all-permissions"/>
+ 			<attribute name="Caller-Allowable-Codebase" value="*.open.ac.uk *.open.edu *.vledev3.open.ac.uk/"/>
```
Comment (sammarshallou) [inline build.xml:196]: Hardcoding this one means nobody outside the OU can use this build script - I don't suppose anyone actually does though so maybe it doesn't matter...

**Review by sammarshallou — CHANGES_REQUESTED** [review]
Thanks, looks basically fine now, a few minor details to consider?

File: src/uk/ac/open/tabapplet/TabApplet.java (hunk 1 above)
Comment (dj4452) [reply src/uk/ac/open/tabapplet/TabApplet.java]: Fixed

File: src/uk/ac/open/tabapplet/TabApplet.java (hunk 2 above)
Comment (dj4452) [reply src/uk/ac/open/tabapplet/TabApplet.java]: Removed this variable.

File: build.xml (hunk 4 above)
Comment (dj4452) [reply build.xml:196]: Have not changed

File: src/uk/ac/open/tabapplet/TabApplet.java (hunk 3 above)
Comment (dj4452) [reply src/uk/ac/open/tabapplet/TabApplet.java]: Moved to EmbeddedRecorderApplet

**Pull request merged**